### ISSUE TITLE
[Bug]: [nightly]Milvus one pod creates, deletes, and creates indexes with errors

### ISSUE THREAD

**Issue by elstic** [milvus-io/milvus#30424]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: nightly-20240128-0542a0e
- Deployment mode(standalone or cluster):standalone
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

case: test_search_cosine_results_same_as_ip 

```
[2024-01-28T20:59:38.583Z] [2024-01-28 20:01:20 - DEBUG - ci_test]: (api_request)  : [Collection.create_index] args: ['float_vector', {'index_type': 'IVF_SQ8', 'params': {'nlist': 64}, 'metric_type': 'COSINE'}, 1200], kwargs: {'index_name': ''} (api_request.py:62)

[2024-01-28T20:59:38.583Z] [2024-01-28 20:01:21 - ERROR - pymilvus.decorators]: RPC error: [create_index], <MilvusException: (code=1, message=447339484649858066: key not found[key=/var/lib/milvus/data/insert_log/447339484649858055/447339484649858056/447339484649858066/104/447339484640679521];)>, <Time:{'RPC start': '2024-01-28 20:01:20.866315', 'RPC error': '2024-01-28 20:01:21.423871'}> (decorators.py:134)
```

fail job: https://jenkins.milvus.io:18080/blue/organizations/jenkins/Milvus%20Nightly%20CI/detail/master/632/pipeline/171

 
 

### Steps To Reproduce

```python
    @pytest.mark.tags(CaseLabel.L2)
    def test_search_cosine_results_same_as_ip(self):
        """
        target: test search results of ip and cosine keep the same
        method: 1. search IP
                2. search cosine
                3. compare the results
        expected: raise no exception
        """
        # 1. create collection and insert data
        collection_w = self.init_collection_general(prefix, True, is_index=False)[0]

        # 2. search IP
        default_index = {"index_type": "IVF_SQ8", "params": {"nlist": 64}, "metric_type": "IP"}
        collection_w.create_index("float_vector", default_index)
        collection_w.load()
        search_params = {"params": {"nprobe": 10}, "metric_type": "IP"}
        res_ip, _ = collection_w.search(vectors[:default_nq], default_search_field,
                                        search_params, default_limit, default_search_exp)

        # 3. search cosine
        collection_w.release()
        collection_w.drop_index()
        default_index = {"index_type": "IVF_SQ8", "params": {"nlist": 64}, "metric_type": "COSINE"}
        collection_w.create_index("float_vector", default_index)
        collection_w.load()
        search_params = {"params": {"nprobe": 10}, "metric_type": "COSINE"}
        res_cosine, _ = collection_w.search(vectors[:default_nq], default_search_field,
                                            search_params, default_limit, default_search_exp)

        # 4. check the search results
        for i in range(default_nq):
            assert res_ip[i].ids == res_cosine[i].ids
```
Steps for reporting errors：
```
[2024-01-28T20:59:38.582Z]         # 3. search cosine

[2024-01-28T20:59:38.582Z]         collection_w.release()

[2024-01-28T20:59:38.582Z]         collection_w.drop_index()

[2024-01-28T20:59:38.582Z]         default_index = {"index_type": "IVF_SQ8", "params": {"nlist": 64}, "metric_type": "COSINE"}

[2024-01-28T20:59:38.582Z] >       collection_w.create_index("float_vector", default_index)

[2024-01-28T20:59:38.582Z] 

[2024-01-28T20:59:38.582Z] testcases/test_search.py:4872: 
```

**Comment by yanliang567** [comment]
/assign @tedxu 
please help to take a look, this is single-pod standalone milvus
/unassign

**Comment by elstic** [comment]
@congqixia 
please take a look at this issue if you have time

**Comment by congqixia** [comment]
@elstic test cases passes locally 
could be fixed by #30666

**Comment by elstic** [comment]
This issue was fixed
Verify image: master-20240311-a99143dd-amd64
